app/controllers/autocomplete_controller.rb

```diff
@@ -88,18 +88,40 @@ def associated_tags
 
   ## NONCANONICAL TAGS
   def noncanonical_tag
-    search_param = params[:term]
+    search_param = Query.new.escape_reserved_characters(params[:term])
     raise "Redshirt: Attempted to constantize invalid class initialize noncanonical_tag #{params[:type].classify}" unless Tag::TYPES.include?(params[:type].classify)
+
     tag_class = params[:type].classify.constantize
-    render_output(tag_class.by_popularity
-                      .where(["canonical = 0 AND name LIKE ?",
-                              '%' + search_param + '%']).limit(10).map(&:name))
+    one_tag = tag_class.find_by(canonical: false, name: params[:term])
```

Comment:
Is there any chance of us regretting this if someone uses a MySQL wildcard character like `%` or `_`?

Reply:
I don't think so; I tried running
```irb
3.1.4 :001 > Tag.find_by(canonical: false, name: '%Stargate%')
 => nil
```
and it returned `nil` pretty fast, so I think there's some escaping going on with `find_by`

Reply:
This is protected by the normal rails protection. 

https://stackoverflow.com/questions/9249624/should-i-escape-the-condition-for-my-rails-find